presto-tpcds/src/test/java/com/facebook/presto/tpcds/statistics/TestTpcdsLocalStats.java

```diff
@@ -58,4 +59,20 @@ public void testTableScanStats()
                         .verifyCharacterColumnStatistics("i_color", absoluteError(0.01)) // nulls fraction estimated as 0
         );
     }
+
+    @Test
+    public void testCharComparison()
+    {
+        // cd_marital_status is char(1)
+        statisticsAssertion.check("SELECT * FROM customer_demographics WHERE cd_marital_status = 'D'",
+                checks -> checks
+                        .estimate(OUTPUT_ROW_COUNT, defaultTolerance())
+                        .estimate(distinctValuesCount("cd_marital_status"), noError()));
+
+        // i_category is char(50)
+        statisticsAssertion.check("SELECT * FROM item WHERE i_category = 'Women                                             '",
+                checks -> checks.estimate(OUTPUT_ROW_COUNT, defaultTolerance()));
+        statisticsAssertion.check("SELECT * FROM item WHERE i_category = cast('Women' as char(50))",
```

Comment:
This case is the simpler case. SFC is indeed involved, but if it wasn't, the result would be the same: we compare symbol reference vs char constant that used to be `$literal$char('...')` (and  `FilterStatsCalculator` delegates 
 to `ScalarStatsCalculator` for everything that isn't plain literal and `SSC` doesn't support function calls) and this didn't work and now it's just `cast('...' as char(50))` and `SSC` handles that...

One could think that `SSC` is the only thing i should be changing (to make it support constant expressions like "magic functions"). Perhaps, but that would not allow `FSC` to recognize true constants (aka literals) and we need this because we have special rules for literals.

BTW Saturated floor cast is what is responsible for converting `i_category = 'Women                                             '` (which in fact is `cast(i_category as varchar(50) = ...`) into `i_category = char 'Women   '`. I.e. it converts the first case (as it is in our tpcds queries) into the second case tested here.